### ISSUE TITLE
Support `Regex` inside `gen` and other functions

### ISSUE THREAD

**Issue by storopoli** [JuliaBooks/Books.jl#305]
It would be great to have a `Regex` support that expands to a vector of `String`s matching the pattern inside `Books.jl` functions.

Here's an example:

```julia
julia> entr(() -> JDS.gen(r"$chapter"), ["contents/"], [JDS])
ERROR: MethodError: no method matching gen(::Regex)
Closest candidates are:
  gen(; call_html, fail_on_error, project, kwargs...) at ~/.julia/packages/Books/nyzNh/src/generate.jl:363
  gen(::AbstractString) at ~/.julia/packages/Books/nyzNh/src/generate.jl:381
  gen(::AbstractString, ::Union{Nothing, Int64}; kwargs...) at ~/.julia/packages/Books/nyzNh/src/generate.jl:381
```

**Comment by rikhuijzer** [comment]
This will be available soon in Books v2.0.4.

Note that you can also use
```julia
julia> entr_gen(r"plots"; M=[JDS])
[...]
```
To save some typing.

**Comment by storopoli** [comment]
Oh.... 🤦‍♂️
